feat(linter): add fix emoji to rules table and doc pages

DonIsaac opened the pull request:
Rules table:
<img width="898" alt="image" src="https://github.com/user-attachments/assets/353052aa-0af3-4c09-8441-ff79f4561ca0">

Doc pages:
<img width="918" alt="image" src="https://github.com/user-attachments/assets/cb43cb2d-15ff-41e6-8523-145cfbc3f484">

DonIsaac commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4715?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4715** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4715?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#4644** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4644?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4640** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4640?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

camc314 commented on crates/oxc_linter/src/fixer/fix.rs:
```diff
@@ -85,6 +85,22 @@ impl FixKind {
+            Self::Suggestion => "💡",
+            Self::Dangerous => "⚠️",
+            _ => {
+                debug_assert!(false, "Please add an emoji for FixKind: {:?}", self);
```
i know this Pr is still a draft, but wouldn't an exhaustive match be better here? it means you can assert this at compile time rather than debug runtime?

DonIsaac replied:
To be exhaustive, I'd need to have branches for all valid `u8` values. However, there are only ~6 valid `FixKind` values.

camc314 replied:
ah gotcha

camc314 commented on tasks/website/src/linter/rules/test.rs:
```diff
@@ -42,9 +42,14 @@ fn parse_type(filename: &str, source_text: &str, source_type: SourceType) -> Res
+    assert!(rendered_html.contains("<table>"));
+    let html = Html::parse_fragment(&rendered_html);
+    assert!(html.errors.is_empty(), "{:#?}", html.errors);
+    // letcHtml::parse_fragment(fragment)
```
```suggestion
```

remove?

camc314 reviewed (approved):
nice stuff!

DonIsaac committed:
feat(linter): add fix emoji to rules table and doc pages (#4715)

Rules table:
<img width="898" alt="image" src="https://github.com/user-attachments/assets/353052aa-0af3-4c09-8441-ff79f4561ca0">

Doc pages:
<img width="918" alt="image" src="https://github.com/user-attachments/assets/cb43cb2d-15ff-41e6-8523-145cfbc3f484">
